hudi-common/src/main/java/org/apache/hudi/common/model/DefaultHoodieRecordPayload.java

```diff
@@ -132,10 +132,10 @@ protected boolean needUpdatingPersistedRecord(IndexedRecord currentValue,
         KeyGeneratorOptions.KEYGENERATOR_CONSISTENT_LOGICAL_TIMESTAMP_ENABLED.defaultValue()));
     Object persistedOrderingVal = HoodieAvroUtils.getNestedFieldVal((GenericRecord) currentValue,
         orderField,
-        true, consistentLogicalTimestampEnabled);
+        false, consistentLogicalTimestampEnabled);
```

Comment:
@wqwl611 Can you please file a JIRA explaining the bug and also add a test case to cover the scenario?